### ISSUE TITLE
Parameter Count of MultiheadAttention does not Change with "num_heads" Argument

### ISSUE THREAD

**Issue by kevin3567** [pytorch/pytorch#60853]
## 🐛 Bug

On torch 1.7.0 the parameter count of torch.nn.MultiheadAttention does not change with "num_heads" argument. As each parallel head is expected to introduce additional parameters, it is suspected that the behaviour indicates a bug. If this is the expected behaviour, please let me know of (explain) the underlying reason or relevant implementation. In the case that this is a bug, please let me know what is the "default" num_heads value in the current implementation (so that I can note down current experimental results). Thanks.

## To Reproduce

a = torch.nn.MultiheadAttention(512, num_heads=1)
sum(p.numel() for p in a.parameters()) # This will return the parameter count 1050624

a = torch.nn.MultiheadAttention(512, num_heads=64)
sum(p.numel() for p in a.parameters()) # This will also return the parameter count 1050624

## Expected behavior

As the number of heads are different, it is expected that the number of parameters should also be different.

## Environment

 - PyTorch Version (e.g., 1.0): 1.7.0 
 - OS (e.g., Linux): Windows and Linux (Ubuntu)
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): pip install torch==1.7.1+cu101 torchvision==0.8.2+cu101 torchaudio==0.7.2 -f https://download.pytorch.org/whl/torch_stable.html
 - Python version: 3.6.7
 - CUDA version: 10.1
 - GPU models and configuration: GeForce RTX 2080
 - Any other relevant information:

## Additional context

cc @jbschlosser @bhosmer @cpuhrsch

**Comment by jbschlosser** [comment]
Hey @kevin3567, the current implementation of MHA divides the embedding dim by the number of heads. This is why the number of parameters stays the same for a fixed embedding dim, even when the number of heads changes.

The reporter of #60445 found this confusing as well. It seems some other implementations (e.g. Keras') multiply the embedding dim by the number of heads instead. This variation would indeed cause the number of parameters to increase as the number of heads is increased.

**Comment by kevin3567** [comment]
Just to make sure I am understanding this correctly, basically:

torch.nn.MultiheadAttention(512, num_heads=1)  # Each parallel head has an embedding size of 512 (512 / 1)
torch.nn.MultiheadAttention(512, num_heads=4)  # Each parallel head has an embedding size of 128 (512 / 4)
torch.nn.MultiheadAttention(512, num_heads=8)  # Each parallel head has an embedding size of 64 (512 / 8)
torch.nn.MultiheadAttention(512, num_heads=64)  # Each parallel head has an embedding size of 8 (512 / 64)

Is that correct?

**Comment by jbschlosser** [comment]
@kevin3567 Yes, that's correct. [Here](https://github.com/pytorch/pytorch/blob/e63db3ae46b707ac724af8dbe9109c7472684288/torch/nn/modules/activation.py#L921) is the section of the code where this is done.

**Comment by kevin3567** [comment]
Okay. thanks
